aws_lambda_powertools/utilities/parser/parser.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from typing import Any, Callable, Dict, Optional
+
+from pydantic import BaseModel, ValidationError
+
+from ...middleware_factory import lambda_handler_decorator
+from ..typing import LambdaContext
+from .envelopes.base import BaseEnvelope
+from .exceptions import InvalidEnvelopeError, InvalidSchemaTypeError, SchemaValidationError
+
+logger = logging.getLogger(__name__)
+
+
+@lambda_handler_decorator
+def parser(
```

Comment:
How would you feel about renaming this to `event_parser`? Naming the module and the contained function the same makes it easier for people to make a mistake and accidentally do `from aws_lambda_powertools.utilities import parser` when they mean `from aws_lambda_powertools.utilities.parser import parser`

Reply:
Good point! Perhaps "...utilities.**parsing** import parser" ?

event_parser wouldn't do justice to its features IMHO, as I can parse any str/obj into classes, enforce type hints at runtime, etc

Reply:
I think event_parser makes sense given that this decorator will always have a lambda event as its input. Happy to disagree and commit though if you prefer renaming the module to parsing! Both are better than having `parser.parser` as the path.

Reply:
Ah yes, didn't consider it from that angle - I'll rename it to event_parser